XYZDriversWebService/src/main/java/com/webservices/xyzdriverswebservice/ClaimEligibility.java

```diff
@@ -0,0 +1,108 @@
+@Stateless()
+public class ClaimEligibility {
+
+    /**
+     * Web service operation
+     */
+    @WebMethod(operationName = "Eligibility")
+    public String Eligibility(
+            @WebParam(name = "username") String username,
+            @WebParam(name = "joinedDate") String joinedDate,
+            @WebParam(name = "listOfClaimDates") ArrayList<String> listOfClaimDates,
+            @WebParam(name = "listOfClaimStatuses") ArrayList<String> listOfClaimStatuses) {
+
+        int claimCounter = 0;
+        Calendar calendar = Calendar.getInstance();
+        Calendar calendar2 = Calendar.getInstance();
+        Calendar calendarJoinedDate = Calendar.getInstance();
+        List<Calendar> calendarClaimDates = new ArrayList();
+        
+        try {
+            SimpleDateFormat sdf = new SimpleDateFormat("yyyy-MM-dd");
+            Date date = sdf.parse(joinedDate);
+            calendarJoinedDate.setTime(date);
+            
+            for(int i = 0; i < listOfClaimDates.size(); i++)
+            {
+                Calendar tempCalendar = Calendar.getInstance();
+                Date tempDate = sdf.parse(listOfClaimDates.get(i));
+                tempCalendar.setTime(tempDate);
+                calendarClaimDates.add(tempCalendar);
+            }
+            
+            NullCheck(username, calendarJoinedDate, listOfClaimDates, listOfClaimStatuses);
+
+            calendar.add(Calendar.MONTH, -6); //six months ago
+
+            if (calendarJoinedDate.after(calendar)) {
+                return username + " has joined less than six months ago. As such, they are not yet eligible to make a claim.";
+            }
+
+            calendar.set(Calendar.YEAR, 2017); //start of current year
+            calendar.set(Calendar.MONTH, Calendar.JANUARY);
+            calendar.set(Calendar.DAY_OF_MONTH, 1);
+            
+            calendar2.set(Calendar.YEAR, 2017); //end of current year
+            calendar2.set(Calendar.MONTH, Calendar.DECEMBER);
+            calendar2.set(Calendar.DAY_OF_MONTH, 31);
+
+            for (int i = 0; i < calendarClaimDates.size(); i++) {
+                if (calendarClaimDates.get(i).after(calendar) && calendarClaimDates.get(i).before(calendar2)) {
+                    if ("APPROVED".equals(listOfClaimStatuses.get(i))) {
+                        claimCounter++;
+                    }
+                }
+            }
+
+            if (claimCounter >= 2) {
+                return username + "has made too many claims this year. As such, they are not yet eligible to make a claim.";
+            }
+
+        } catch (NullPointerException | ParseException ex) {
+            return ex.toString();
+        }
+
+        return username + " is eligible to make a claim! They have made " + claimCounter + " claims this year.";
+    }
+
+    private void NullCheck(String username, Calendar joinedDate, ArrayList listOfClaimDates, ArrayList listOfClaimStatuses) throws NullPointerException {

```

Comment:
You shouldn't throw NullPointerException, this is an exception you'd only really expect to be thrown by the runtime itself (goes for C# too). It suggests that a null pointer was dereferenced and that hasn't happened here so you've broken the semantics.

Since you're checking whether or not arguments passed to the service are null you could throw IllegalArgumentException instead. You can catch this instead of NullPointerException.